File: travelAgency/Model/src/main/java/app/model/dto/SearchFlightsDto.java
package app.model.dto;

import java.io.Serializable;
import java.time.LocalDate;

public class SearchFlightsDto implements Serializable {
    private String destination;
    private LocalDate date;

    public SearchFlightsDto(String destination, LocalDate date) {
        this.destination = destination;
        this.date = date;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return "SearchFlightsDto{" +
                "destination='" + destination + '\'' +
                ", date=" + date +
                '}';
    }
}

File: Anul_2/Semestrul_2/Medii de proiectare si programare/mpp-proiect-csharp-andrea-muresan/TravelAgencyProject/Model/dto/SearchFlightsDto.cs
using System;
using System.Collections.Generic;

namespace Model.dto
{
    [Serializable]
    public class SearchFlightsDto
    {
        public string Destination { get;}
        public string Date { get; }

        public SearchFlightsDto(string destination, string date)
        {
            Destination = destination;
            Date = date;
        }
        
        public override bool Equals(object obj)
        {
            return obj is SearchFlightsDto dto &&
                   Date == dto.Date &&
                   Destination == dto.Destination;
        }

        public override int GetHashCode()
        {
            int hashCode = 568732665;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Destination);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Date);
            return hashCode;
        }
    }
}